src/iterator.jl

```diff
@@ -43,6 +43,14 @@ function Base.hash(iter::TestIterator{<:Any,IT,IS}) where {IT,IS}
     return mapreduce(hash, hash, (iter.data, IT, IS))
 end
 
+# To make it a valid differential: needs at very least `zero` and `+`
+Base.zero(::Type{<:TestIterator}) = Zero()
+function Base.:+(iter1::TestIterator{T,IS,IE}, iter2::TestIterator{T,IS,IE}) where {T,IS,IE}
+    return TestIterator{T,IS,IE}(map(+, iter1, iter2))
+end
```

Comment:
We were using this like a differential, without these being defined,
which ment that testing accumulation errored.